File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    tmp = prices[::-1]
    
    while tmp:
        price = tmp.pop()
        sec = 0;
        for i in range(len(tmp)-1, -1, -1):
            sec += 1
            if price > tmp[i]: break
        answer.append(sec)
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    let answer = new Array(prices.length).fill(0);
    let stack = []
    
    for (let i = 0; i < prices.length; i++){
        while (stack.length && stack[stack.length-1][1] > prices[i]){
            let [past, _] = stack.pop();
            answer[past] = i - past;
        }
        stack.push([i,prices[i]])
    }
    
    for (let [i, _] of stack) answer[i] = prices.length - 1 - i
    
    return answer;
}